TST: windows failures

```
======================================================================
FAIL: test_c_engine (pandas.io.tests.parser.test_unsupported.TestUnsupportedFeatures)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\conda\Documents\pandas3.5\pandas\io\tests\parser\test_unsupported.py", line 64, in test_c_engine
    read_table(StringIO(data), engine='c', sep='\xa7')
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 2437, in __exit__
    raise AssertionError("{0} not raised.".format(name))
AssertionError: ValueError not raised.
```

closed in #14140 
```
======================================================================
FAIL: test_append_zero (pandas.sparse.tests.test_list.TestSparseList)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\conda\Documents\pandas3.5\pandas\sparse\tests\test_list.py", line 64, in test_append_zero
    tm.assert_sp_array_equal(sparr, SparseArray(arr, fill_value=0))
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1392, in assert_sp_array_equal
    assert_numpy_array_equal(left.sp_values, right.sp_values)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1083, in assert_numpy_array_equal
    assert_attr_equal('dtype', left, right, obj=obj)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 878, in assert_attr_equal
    left_attr, right_attr)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different

Attribute "dtype" are different
[left]:  int64
[right]: int32

======================================================================
FAIL: test_dataframe_dummies_prefix_str (pandas.tests.test_reshape.TestGetDummies)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\conda\Documents\pandas3.5\pandas\tests\test_reshape.py", line 327, in test_dataframe_dummies_prefix_str
    assert_frame_equal(result, expected)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1313, in assert_frame_equal
    obj='DataFrame.iloc[:, {0}]'.format(i))
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1154, in assert_series_equal
    assert_attr_equal('dtype', left, right)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 878, in assert_attr_equal
    left_attr, right_attr)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: Attributes are different

Attribute "dtype" are different
[left]:  int64
[right]: int32

======================================================================
FAIL: test_dataframe_dummies_prefix_str (pandas.tests.test_reshape.TestGetDummiesSparse)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\conda\Documents\pandas3.5\pandas\tests\test_reshape.py", line 327, in test_dataframe_dummies_prefix_str
    assert_frame_equal(result, expected)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1313, in assert_frame_equal
    obj='DataFrame.iloc[:, {0}]'.format(i))
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1154, in assert_series_equal
    assert_attr_equal('dtype', left, right)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 878, in assert_attr_equal
    left_attr, right_attr)
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 1018, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: Attributes are different

Attribute "dtype" are different
[left]:  int64
[right]: int32
```

This replicates on 3.5 windows for me:

```
======================================================================
FAIL: test_c_engine (pandas.io.tests.parser.test_unsupported.TestUnsupportedFeatures)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\conda\Documents\pandas3.5\pandas\io\tests\parser\test_unsupported.py", line 64, in test_c_engine
    read_table(StringIO(data), engine='c', sep='\xa7')
  File "C:\Users\conda\Documents\pandas3.5\pandas\util\testing.py", line 2450, in __exit__
    raise AssertionError("{0} not raised.".format(name))
AssertionError: ValueError not raised.

----------------------------------------------------------------------
Ran 10873 tests in 433.840s

FAILED (SKIP=168, failures=1)

(pandas3.5) C:\Users\conda\Documents\pandas3.5>ipyton
'ipyton' is not recognized as an internal or external command,
operable program or batch file.

(pandas3.5) C:\Users\conda\Documents\pandas3.5>ipython
Python 3.5.1 |Continuum Analytics, Inc.| (default, Feb 16 2016, 09:49:46) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 5.1.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: pd.show_versions()
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-eaa3460ab7ab> in <module>()
----> 1 pd.show_versions()

NameError: name 'pd' is not defined

In [2]: import pandas as pd

In [3]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: 59524af1e90d3dda2d885e711f7258704a020b6d
python: 3.5.1.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 70 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.18.1+427.g59524af
```

I think the encoding is getting set to windows default

```
In [5]: sys.getdefaultencoding()
Out[5]: 'utf-8'
```

which seems ok (I though ``ascii``) would trigger it.
I'd kind of like to keep this tests now that it shows on Travis as well (you can rebase off that xref PR, which shows on Travis to test if that helps).

--- a/pandas/io/parsers.py
+++ b/pandas/io/parsers.py
@@ -800,17 +800,22 @@ class TextFileReader(BaseIterator):
                                   " different from '\s+' are"\
                                   " interpreted as regex)"
                 engine = 'python'
-
-        elif len(sep.encode(encoding)) > 1:
-            if engine not in ('python', 'python-fwf'):
-                fallback_reason = "the separator encoded in {encoding}"\
-                                  " is > 1 char long, and the 'c' engine"\
-                                  " does not support such separators".format(
-                                      encoding=encoding)
-                engine = 'python'
         elif delim_whitespace:
             if 'python' in engine:
                 result['delimiter'] = '\s+'
+        elif sep is not None:
+            encodeable = True
+            try:
+                if len(sep.encode(encoding)) > 1:
+                    encodeable = False
+            except UnicodeDecodeError:
+                encodeable = False
+            if not encodeable and engine not in ('python', 'python-fwf'):
+                fallback_reason = "the separator encoded in {encoding}" \
+                                  " is > 1 char long, and the 'c' engine" \
+                                  " does not support such separators".format(
+                                      encoding=encoding)
+                engine = 'python'
 
         if fallback_reason and engine_specified:
             raise ValueError(fallback_reason)

--- a/pandas/io/tests/parser/test_unsupported.py
+++ b/pandas/io/tests/parser/test_unsupported.py
@@ -60,10 +60,6 @@ class TestUnsupportedFeatures(tm.TestCase):
                        sep=None, delim_whitespace=False)
         with tm.assertRaisesRegexp(ValueError, msg):
             read_table(StringIO(data), engine='c', sep='\s')
-
-        # GH 14120, skipping as failing when locale is set
-        # with tm.assertRaisesRegexp(ValueError, msg):
-        #     read_table(StringIO(data), engine='c', sep='§')
         with tm.assertRaisesRegexp(ValueError, msg):
             read_table(StringIO(data), engine='c', skipfooter=1)
 

